### ISSUE TITLE
Failed to read response from server: 20 (and  go to definition not working)

### ISSUE THREAD

**Issue by liuzqt** [microsoft/vscode-cpptools#6214]
**Type: LanguageService**

**Describe the bug**
- OS and Version: local macOS 10.15.16, connect to remote Ubuntu 16.04
- VS Code Version: 1.49.2
- C/C++ Extension Version: 1.0.1
- Other extensions you installed (and if the issue persists after disabling them):
- Does this issue involve using SSH remote to run the extension on a remote machine?: yes
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

**Steps to reproduce**
(I'm connecting to remote machine using SSH remote)
reload the window
open a cpp file, click on a symbol and go the definition/declaration, showing definition not found

the strange thing is , when I hover over the symbol, it can show the signature correctly, so in some extend it's working, but just not able to jump to the right place.

**Expected behavior**
should be able to jump to definition/declaration

<details>
  <summary><strong>Logs</strong></summary>

```

Failed to read response from server: 20
```
</details>

**Screenshots**

<img width="2114" alt="截屏2020-09-25 下午2 33 49" src="https://user-images.githubusercontent.com/22358241/94317975-53f66a00-ff3c-11ea-9cc8-148c00992582.png">
<img width="1483" alt="截屏2020-09-25 下午2 33 59" src="https://user-images.githubusercontent.com/22358241/94317981-5658c400-ff3c-11ea-8ce5-027f8e9c63eb.png">
<img width="1132" alt="截屏2020-09-25 下午2 35 46" src="https://user-images.githubusercontent.com/22358241/94318017-6f617500-ff3c-11ea-8c0d-ede378de3433.png">

**Additional context**

**Comment by sean-mcmanus** [comment]
The cpptools-srv process is crashing. Are you able to provide a repro? Or get a crash call stack by attaching to cpptools-srv: https://github.com/microsoft/vscode-cpptools/wiki/Attaching-debugger-to-cpptools-or-cpptools%E2%80%90srv

**Comment by sean-mcmanus** [comment]
Also, you can set C_Cpp.loggingLevel to "Debug" to get more info about which TU is being launched and causing the crash.

**Comment by liuzqt** [comment]
I solved the issue by reboot my remote machine....

When I tried to attach cpptools-srv, I found there're several cpptool-srv processes, and even mixed of different version. Frankly speaking I haven't clean up my remote machine for a while, so I do a reboot and everything goes well now.

A small question, how is it possible that my remote machines running many cpptools/cpptools-srv proceses? Do I need to manually clean them after I close my remote-ssh session?
